# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dvdstyler/x86_64.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -ur DVDStyler-1.4_orig/src/ButtonPropDlg.cpp DVDStyler-1.4/src/ButtonPropDlg.cpp
--- DVDStyler-1.4_orig/src/ButtonPropDlg.cpp	2006-08-25 14:06:34.000000000 +0100
+++ DVDStyler-1.4/src/ButtonPropDlg.cpp	2006-08-25 14:20:22.000000000 +0100
@@ -61,7 +61,7 @@
   if (!customAction.length())
   {
 	// check if selection was ok, otherwise set this action as custom action
-	int tsi = m_object->GetActionTsi() > -2 ? m_object->GetActionTsi() : m_tsi;
+	long tsi = m_object->GetActionTsi() > -2 ? m_object->GetActionTsi() : m_tsi;
 	if (GetTsi() != tsi || GetPgci() != m_object->GetActionPgci() ||
 		GetChapter() != m_object->GetActionChapter())
 	  customAction = m_object->GetAction();
@@ -112,21 +112,21 @@
   sizer->Add(mainSizer, 1, wxEXPAND|wxALL, 10);
 }
 
-int ButtonPropDlg::GetTsi()
+long ButtonPropDlg::GetTsi()
 {
   if (m_titlesetChoice->GetSelection() >= 0)
-	return ((int) m_titlesetChoice->GetClientData(m_titlesetChoice->GetSelection()))-1;
+	return ((long) m_titlesetChoice->GetClientData(m_titlesetChoice->GetSelection()))-1;
   return 0;
 }
 
-int ButtonPropDlg::GetPgci()
+long ButtonPropDlg::GetPgci()
 {
   if (m_targetChoice->GetSelection() >= 0)
-	return (int) m_targetChoice->GetClientData(m_targetChoice->GetSelection());
+	return (long) m_targetChoice->GetClientData(m_targetChoice->GetSelection());
   return 0;
 }
 
-int ButtonPropDlg::GetChapter()
+long ButtonPropDlg::GetChapter()
 {
   if (m_chapterChoice->GetSelection() >= 0)
 	return m_chapterChoice->GetSelection();
@@ -137,7 +137,7 @@
 {
   int sel = 0;
   wxChoice* ctrl = m_titlesetChoice;
-  int tsi = m_object->GetActionTsi() > -2 ? m_object->GetActionTsi() : m_tsi;
+  long tsi = m_object->GetActionTsi() > -2 ? m_object->GetActionTsi() : m_tsi;
   ctrl->Clear();
   if (m_dvd->GetVmgm().GetCount()>0)
 	ctrl->Append(_("video manager"), (void*) 0);
@@ -157,8 +157,8 @@
 {
   int sel = 0;
   wxChoice* ctrl = m_targetChoice;
-  int tsi = GetTsi();
-  int pgci = ctrl->GetCount() ? GetPgci() : m_object->GetActionPgci();
+  long tsi = GetTsi();
+  long pgci = ctrl->GetCount() ? GetPgci() : m_object->GetActionPgci();
   ctrl->Clear();
   PgcArray& pgcs = m_dvd->GetPgcArray(tsi, true);
   for (int i=0; i<(int)pgcs.Count(); i++)
@@ -185,9 +185,9 @@
 void ButtonPropDlg::UpdateChapters()
 {
   wxChoice* ctrl = m_chapterChoice;
-  int tsi = GetTsi();
-  int pgci = GetPgci();
-  int chapter = ctrl->GetCount() ? GetChapter() : m_object->GetActionChapter();
+  long tsi = GetTsi();
+  long pgci = GetPgci();
+  long chapter = ctrl->GetCount() ? GetChapter() : m_object->GetActionChapter();
   ctrl->Clear();
   ctrl->Append(wxString::Format(_("chapter %d"), 1));
   if (tsi>=0 && pgci%2 == 1)
@@ -296,9 +296,9 @@
 	return false;
   }
   int n = 1;
-  int tsi = ((int)GetClientData(n++))-1;
+  long tsi = ((long)GetClientData(n++))-1;
   m_object->SetActionTsi(tsi != m_tsi ? tsi : -2);
-  m_object->SetActionPgci((int)GetClientData(n++));
+  m_object->SetActionPgci((long)GetClientData(n++));
   m_object->SetActionChapter(GetInt(n++));
   if (GetBool(n++))
 	m_object->SetCustomAction(GetString(n));
diff -ur DVDStyler-1.4_orig/src/ButtonPropDlg.h DVDStyler-1.4/src/ButtonPropDlg.h
--- DVDStyler-1.4_orig/src/ButtonPropDlg.h	2006-08-25 14:06:34.000000000 +0100
+++ DVDStyler-1.4/src/ButtonPropDlg.h	2006-08-25 14:20:24.000000000 +0100
@@ -31,17 +31,17 @@
     MenuButton* m_object;
 	Menu* m_menu;
 	DVD* m_dvd;
-	int m_tsi;
-	int m_pgci;
-	int m_btInd;
+	long m_tsi;
+	long m_pgci;
+	long m_btInd;
 	void CreatePropPanel(wxSizer* sizer);
 	
 	wxChoice* m_titlesetChoice;
 	wxChoice* m_targetChoice;
 	wxChoice* m_chapterChoice;
-	int GetTsi(); // returns selected titleset id
-	int GetPgci(); // returns selected pgc (target) id
-	int GetChapter(); // returns selected chapter nummer
+	long GetTsi(); // returns selected titleset id
+	long GetPgci(); // returns selected pgc (target) id
+	long GetChapter(); // returns selected chapter nummer
 	void UpdateTitlesets();
 	void UpdateTargets();
 	void UpdateChapters();
